src/trace_link/kineto_operator.py

```diff
@@ -153,3 +155,12 @@ def is_gpu_op(self) -> bool:
         """
         gpu_categories = {"kernel", "gpu_memcpy"}
         return self.category in gpu_categories
+
+    def is_collective_op(self) -> bool:
+        """
+        Check if the operator is a GPU-side collective operator based on its name.
+
+        Returns
+            bool: True if it's a GPU-side collective, otherwise False.
+        """
+        return "ncclDevKernel" in self.name
```

Comment:
Having ncclDevKernel in the name does not mean that it is a collective communication operator. It can be an NCCL communication operator, but it does not guarantee that it is a collective communication operator.
```
grep "ncclDevKernel" ~/Downloads/gpt3_126m_1.1.0-chakra.0.0.4/kineto_0.json | grep -v AllReduce
    "ph": "X", "cat": "kernel", "name": "ncclDevKernel_SendRecv(ncclDevKernelArgsStorage<4096ul>)", "pid": 0, "tid": 64,
    "ph": "X", "cat": "kernel", "name": "ncclDevKernel_SendRecv(ncclDevKernelArgsStorage<4096ul>)", "pid": 0, "tid": 64,
    "ph": "X", "cat": "kernel", "name": "ncclDevKernel_SendRecv(ncclDevKernelArgsStorage<4096ul>)", "pid": 0, "tid": 64,
    "ph": "X", "cat": "kernel", "name": "ncclDevKernel_SendRecv(ncclDevKernelArgsStorage<4096ul>)", "pid": 0, "tid": 64,
```

Reply:
Just noticed that, ncclDevKernel_SendRecv is not considered as collective communication. 
Though it should have `pg_name`, this is point-to-point send/receive primitives. 
According to this document (https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/overview.html), I will remanme it 'is_inter_gpu_communication()'.